api/src/test/java/org/openmrs/module/locationbasedaccess/aop/PatientSearchAdviserTest.java

```diff
@@ -0,0 +1,212 @@
+        assertTrue(authenticatedUser.isSuperUser());
+
+        Location location1 = locationService.getLocation(DEMO_LOCATION1_ID);
+        Location location2 = locationService.getLocation(DEMO_LOCATION2_ID);
+
+        List<Patient> patientList = patientService.getPatients("Patient");
+        assertEquals(patientList.size(), 3);
+        assertEquals(patientList.get(0).getAttribute(DEMO_PERSON_ATTRIBUTE_TYPE_ID).getValue(), location1.getUuid());
+        assertEquals(patientList.get(1).getAttribute(DEMO_PERSON_ATTRIBUTE_TYPE_ID).getValue(), location2.getUuid());
+        assertNull(patientList.get(2).getAttribute(DEMO_PERSON_ATTRIBUTE_TYPE_ID));
+    }
+
+    @Test
+    public void getPatient_adminShouldGetPatientsFromAllLocations() {
+        Context.authenticate("admin", "test");
+        User authenticatedUser = Context.getAuthenticatedUser();
+        assertNotNull(authenticatedUser);
+        assertTrue(authenticatedUser.isSuperUser());
+
+        Location location1 = locationService.getLocation(DEMO_LOCATION1_ID);
+        Location location2 = locationService.getLocation(DEMO_LOCATION2_ID);
+
+        Patient patient1 = patientService.getPatient(2);
+        assertNotNull(patient1);
+        assertNull(patient1.getAttribute(DEMO_PERSON_ATTRIBUTE_TYPE_ID));
+
+        Patient patient2 = patientService.getPatient(3);
+        assertNotNull(patient2);
+        assertEquals(patient2.getAttribute(DEMO_PERSON_ATTRIBUTE_TYPE_ID).getValue(), location1.getUuid());
+
+        Patient patient3 = patientService.getPatient(4);
+        assertNotNull(patient3);
+        assertEquals(patient3.getAttribute(DEMO_PERSON_ATTRIBUTE_TYPE_ID).getValue(), location2.getUuid());
+    }
+
+    @Test
+    public void getPatientByUuid_adminShouldGetPatientsFromAllLocations() {
+        Context.authenticate("admin", "test");
+        User authenticatedUser = Context.getAuthenticatedUser();
+        assertNotNull(authenticatedUser);
+        assertTrue(authenticatedUser.isSuperUser());
+
+        Location location1 = locationService.getLocation(DEMO_LOCATION1_ID);
+        Location location2 = locationService.getLocation(DEMO_LOCATION2_ID);
+
+        Patient patient1 = patientService.getPatientByUuid("5631b434-78aa-102b-91a0-001e378eb67e");
+        assertNotNull(patient1);
+        assertNull(patient1.getAttribute(DEMO_PERSON_ATTRIBUTE_TYPE_ID));
+
+        Patient patient2 = patientService.getPatientByUuid("5631b434-78aa-102b-91a0-001e378eb68e");
+        assertNotNull(patient2);
+        assertEquals(patient2.getAttribute(DEMO_PERSON_ATTRIBUTE_TYPE_ID).getValue(), location1.getUuid());
+
+        Patient patient3 = patientService.getPatientByUuid("5631b434-78aa-102b-91a0-001e378eb69e");
+        assertNotNull(patient3);
+        assertEquals(patient3.getAttribute(DEMO_PERSON_ATTRIBUTE_TYPE_ID).getValue(), location2.getUuid());
+    }
+
+
+    @Test
+    public void getPatients_nonAdminUserWithOutLocationPropertyShouldNotGetPatientInformation() {
```

Comment:
Is this a test for getPatients, getPatient or getPatientByUuid

Reply:
But i still see those extra method calls in this test.

Reply:
Sorry I didn't get you. I just tested the getPatients() inside this test method. See [here](https://github.com/openmrs/openmrs-module-locationbasedaccess/pull/38/files/a856f3d988a2fefb27618dfa9573d34706e5eef9#diff-ba475a8a0e81f093e143aa265716d13bR62)

Reply:
I did not expect to still see getPatient or getPatientByUuid in this test.

Reply:
I just changed the code @dkayiwa  please have a look again